### PR TITLE
Fix for Metrics & Logging pv-pvc crisscross problem

### PR DESCRIPTION
**Pull request by mazzystr**

### CODE REVIEW

File: roles/openshift_persistent_volumes/templates/persistent-volume.yml.j2 [hunk 1]
```diff
@@ -7,6 +7,8 @@ items:
   kind: PersistentVolume
   metadata:
     name: "{{ volume.name }}"
+    labels:
+      name: "{{ volume.name }}"
```
Comment (ewolinetz) [inline roles/openshift_persistent_volumes/templates/persistent-volume.yml.j2]: can we make this more generic instead? i think we should be able to provide any key and value for a label

File: roles/openshift_metrics/templates/pvc.j2 [hunk 2]
```diff
@@ -25,3 +25,8 @@ spec:
   resources:
     requests:
       storage: {{size}}
+{% if openshift_metrics_cassandra_pvc_prefix is defined and openshift_metrics_cassandra_pvc_prefix %}
+  selector:
+    matchLabels:
+      name: {{ openshift_metrics_cassandra_pvc_prefix }}
```
Comment (ewolinetz) [inline roles/openshift_metrics/templates/pvc.j2]: can we make this more generic, similar to logging?

File: roles/openshift_logging/tasks/generate_pvcs.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@
     obj_name: "{{claim_name}}"
     size: "{{es_pvc_size}}"
     access_modes: "{{ es_access_modes | list }}"
-    pv_selector: "{{es_pv_selector}}"
+    pv_selector: "{{es_pv_selector | default(['name': openshift_hosted_logging_storage_volume_name])}}"
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/generate_pvcs.yaml]: if we can correctly provide a label to use when generating the PV we shouldnt need to use this as a default...

File: filter_plugins/oo_filters.py [hunk 4]
```diff
@@ -749,7 +749,7 @@ def oo_persistent_volumes(hostvars, groups, persistent_volumes=None):
                         size = params['volume']['size']
                         access_modes = params['access']['modes']
                         persistent_volume = dict(
-                            name="{0}-volume".format(volume),
+                            name="{0}".format(volume),
```
Comment (ewolinetz) [inline filter_plugins/oo_filters.py]: can we make this a separate PR? I think we should evaluate dropping this and the potential impact it will have with existing systems/customers

**Comment by mazzystr** [comment]
Sorry guys.  I'm taken off the RHMAP / OpenShift project

**Comment by ewolinetz** [comment]
@mazzystr mind if we close this in favour of https://github.com/openshift/openshift-ansible/pull/4302?
